SignalServiceKit/src/Util/OWS2FAManager.m

```diff
@@ -65,16 +65,30 @@ - (instancetype)initWithStorageManager:(TSStorageManager *)storageManager
     return self;
 }
 
+- (nullable NSString *)pin
+{
+    return [self.dbConnection objectForKey:kOWS2FAManager_PINKey inCollection:kOWS2FAManager_Collection];
+}
+
 - (BOOL)is2FAEnabled
 {
-    return [self.dbConnection boolForKey:kOWS2FAManager_IsEnabledKey
-                            inCollection:kOWS2FAManager_Collection
-                            defaultValue:NO];
+    return [self.dbConnection hasObjectForKey:kOWS2FAManager_PINKey inCollection:kOWS2FAManager_Collection];
+}
+
+- (void)set2FANotEnabled
+{
+    [self.dbConnection removeObjectForKey:kOWS2FAManager_PINKey inCollection:kOWS2FAManager_Collection];
+
+    [[NSNotificationCenter defaultCenter] postNotificationNameAsync:NSNotificationName_2FAStateDidChange
+                                                             object:nil
+                                                           userInfo:nil];
 }
 
-- (void)setIs2FAEnabled:(BOOL)value
+- (void)set2FAEnabledWithPin:(NSString *)pin
```

Comment:
I had to make similar changes in #3085 - can you use those?

Reply:
Rebased and reconciled.  I drew from both but crucially we now have a single piece of state - a "registration lock" pin.  If it's set, "registration lock" is enabled.  Less state = less surface area for bugs.